scripts/docs/generate_asset_documentation.py

```diff
@@ -10,184 +10,305 @@
+
+    @property
+    def doc(self):
+        _doc = snakemd.new_doc()
+        return _doc
+
+    def save(self):
+        self.directory.mkdir(exist_ok=True)
+        with open(self.fullpath, 'w') as f:
+            f.write(str(self.doc))
+
+    @staticmethod
+    def create_asset_info(asset_config):
+        # TODO: Use AssetType.COMPONENT
+        if asset_config.type.value == "environment":
+            return EnvironmentInfo(asset_config)
+        if asset_config.type.value == "component":
+            return ComponentInfo(asset_config)
+        if asset_config.type.value == "model":
+            return ModelInfo(asset_config)
+
+        raise Exception(f"Not supported asset type {asset_config.type}. Use {supported_asset_types}")
+
+# region Doc Formatting
+    def _add_doc_name(self, doc):
+        if "display_name" in self._asset:
+            doc.add_heading(self._asset['display_name'])
+        doc.add_heading(self.name)
+
+    def _add_doc_overview(self, doc):
+        doc.add_heading("Overview ", level=2)
+
+    def _add_doc_description(self, doc):
+        doc.add_paragraph("**Description**: " + self.description)
+
+    def _add_doc_asset_version(self, doc):
+        doc.add_paragraph("**Version**: " + self.version)
+
+    def _add_doc_license_from_tags(self, doc):
+        if "tags" in self._asset and "license" in self._asset["tags"]:
+            doc.add_paragraph("**License**: " + self._asse["tags"]["license"])
+
+    def _add_doc_properties(self, doc):
+        if "properties" in self._asset:
+            doc.add_heading("Properties", level=4)
+            for property, value in self._asset["properties"].items():
+                doc.add_paragraph(f"**{property}**: {value}")
+
+    def _add_doc_tags(self, doc):
+        if "tags" in self._asset:
+            doc.add_heading("Tags ", level=3)
+            tags = []
+            for tag, value in self._asset["tags"].items():
+                tags.append("`{}` ".format(tag if len(str(value)) == 0 else (tag + " : " + str(value))))
+            tags.sort(key=lambda x: x[0].lower())
+            doc.add_raw(" ".join(tags))
+        return doc
+
+    def _add_doc_mcr_image(self, doc):
+        """Add MCR Image."""
+        img = ":".join([self._asset_config.extra_config_as_object().get_full_image_name(), self.version])
+        doc.add_paragraph("**Docker image**: " + img)
+
+    def _add_doc_docker_context(self, doc):
+        """Add docker context content."""
+        doc.add_heading("Docker build context", level=2)
+        context = {}
+        dockerfile = ""
+        for context_file in search(str(self._asset_config.extra_config_as_object().context_dir_with_path) + "/*"):
+            content = ""
+            with open(context_file, "r") as f:
+                # do not expect nested structure for now
+                content = f.read()
+            filename = os.path.basename(context_file)
+            if filename.lower() == "dockerfile":
```

Comment:
Please consider using _asset_config.extra_config_as_object()'s (EnvironmentConfig's) dockerfile properties.

Reply:
that's in my todo from your comment to a previous PR :)